### DISCUSSION TITLE
Hamlet SignalK problem

### DISCUSSION THREAD

**Discussion by rodegerdts** [hatlabs/discussions#131] [Q&A]
Hi
I have a problem with Hamlet not communicating with the Signal K server. The authentication process works as expected and the server shows Hamlet with read/write permissions. According to the SenseESP documentation data should be showing up on the server but it does not. NMEA2000 works just fine. The server is a Raspi5 running Openplotter 4. SignalK v2.9.0. I am pasting the monitor output below som reference. I hope someone can help me here.

(connect)(C0) Initiating websocket connection with server...
(get_mdns_service)(C0) Found server 192.168.0.70 (port 3000)
(connect)(C0) Signal K server has been found at address 192.168.0.70:3000 by mDNS.
(connect)(C1) Websocket is connecting to Signal K server on address 192.168.0.70:3000
(test_token)(C0) Testing token with url http://192.168.0.70:3000/signalk/v1/stream
(test_token)(C0) Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkZXZpY2UiOiIxNTczOTJiYS1iMDNiLTA2YTMtODA5Ny01YTM2YTA5Yzk4NWYiLCJpYXQiOjE3MjUyMTc5MTF9.a7jyG-bcUNLmmQIdkvkqq9OlSBaPh61c0Cm9RVb1z2c
(test_token)(C0) Testing resulted in http status 426
(test_token)(C0) Returned payload (length 16) is: 
(test_token)(C0) Upgrade Required
(test_token)(C0) Attempting to connect to Signal K Websocket...
(on_connected)(C0) Websocket client connected to URL: /signalk/v1/stream?subscribe=none

(on_connected)(C0) Subscribing to Signal K listeners...
(operator())(C1) Got system status: 106

**Comment by kristianschneider** [comment]
I face a very similar issue. 
Any thoughts @mairas ?

**Comment by kristianschneider** [comment]
> Hi I have a problem with Hamlet not communicating with the Signal K server. The authentication process works as expected and the server shows Hamlet with read/write permissions. According to the SenseESP documentation data should be showing up on the server but it does not. NMEA2000 works just fine. The server is a Raspi5 running Openplotter 4. SignalK v2.9.0. I am pasting the monitor output below som reference. I hope someone can help me here.
> 
> (connect)(C0) Initiating websocket connection with server... (get_mdns_service)(C0) Found server 192.168.0.70 (port 3000) (connect)(C0) Signal K server has been found at address 192.168.0.70:3000 by mDNS. (connect)(C1) Websocket is connecting to Signal K server on address 192.168.0.70:3000 (test_token)(C0) Testing token with url http://192.168.0.70:3000/signalk/v1/stream (test_token)(C0) Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkZXZpY2UiOiIxNTczOTJiYS1iMDNiLTA2YTMtODA5Ny01YTM2YTA5Yzk4NWYiLCJpYXQiOjE3MjUyMTc5MTF9.a7jyG-bcUNLmmQIdkvkqq9OlSBaPh61c0Cm9RVb1z2c (test_token)(C0) Testing resulted in http status 426 (test_token)(C0) Returned payload (length 16) is: (test_token)(C0) Upgrade Required (test_token)(C0) Attempting to connect to Signal K Websocket... (on_connected)(C0) Websocket client connected to URL: /signalk/v1/stream?subscribe=none
> 
> (on_connected)(C0) Subscribing to Signal K listeners... (operator())(C1) Got system status: 106

@rodegerdts did you have any luck with getting further ?

**Reply by rodegerdts** [reply]
No. Not really. As I understand this issue the hamlet firmware needs an update. I tried to install the senseESP firmware but this is not working either. Maybe I will just have to use this as a NMEA2000 device. This is a pity since SignalK is supposed to be able to replace NMEA2000. Unfortunately I am just not a good enough programmer to solve this problem myself.

**Reply by kristianschneider** [reply]
There is a new example firmware that works
